### PR TITLE
[bitnami/kafka] Setup of sasl authentication to external zookeeper in server.properties not working (bitnami#22729)

### LINKED ISSUE #22729: [bitnami/kafka] Setup of sasl authentication to external zookeeper in server.properties not working
### Name and Version

bitnami/kafka

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Upgrading Helm-Chart from 23.0.7 to 26.6.2

```yaml
kraft:
  enabled: false

broker:
  replicaCount: 5
  zookeeperMigrationMode: false

controller:
  replicaCount: 0

zookeeper:
  enabled: false
externalZookeeper:
  servers=[ip-1, ip-2, ip-3 ...]

listeners:
  client:
    protocol: SSL
  interbroker:
    protocol: SSL
  external:
    protocol: SSL

sasl:
  zookeeper:
    user: zookeeper-user
  existingSecret: name-of-the-secret

extraConfig:
...
  authorizer.class.name=kafka.security.authorizer.AclAuthorizer
...
```
The secret of sasl.existingSecret has a key "zookeeper-password".
All keystores are setup properly.

### What is the expected behavior?

Kafka brokers connect to external zookeeper through sasl username and password.

### What do you see instead?

Kafka brokers do not connect to external zookeeper through sasl username and password and shutdown.

### Additional information

After startup the server.properties file of each broker includes these elements:

```
sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required
    username="zookeeper-user"
    password="zookeeper-password-placeholder";
```
generated at https://github.com/bitnami/charts/blob/main/bitnami/kafka/templates/_helpers.tpl#L679-L683

```
{{- if .Values.sasl.zookeeper.user }}
sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \
    username="{{ .Values.sasl.zookeeper.user }}" \
    password="zookeeper-password-placeholder";
{{- end }}
```

**Problem 1**:
The password placeholder is not replaced with proper password due to `{{- if (include "kafka.saslEnabled" .) }}` in https://github.com/bitnami/charts/blob/main/bitnami/kafka/templates/scripts-configmap.yaml#L286-L316

```
    {{- if (include "kafka.saslEnabled" .) }}
    configure_kafka_sasl() {

      # Replace placeholders with passwords
      {{- if regexFind "SASL" (upper .Values.listeners.interbroker.protocol) }}
      {{- if (include "kafka.saslUserPasswordsEnabled" .) }}
      replace_placeholder "interbroker-password-placeholder" "$KAFKA_INTER_BROKER_PASSWORD"
      {{- end }}
      {{- if (include "kafka.saslClientSecretsEnabled" .) }}
      replace_placeholder "interbroker-client-secret-placeholder" "$KAFKA_INTER_BROKER_CLIENT_SECRET"
      {{- end }}
      {{- end -}}
      {{- if and .Values.kraft.enabled (regexFind "SASL" (upper .Values.listeners.controller.protocol)) }}
      {{- if (include "kafka.saslUserPasswordsEnabled" .) }}
      replace_placeholder "controller-password-placeholder" "$KAFKA_CONTROLLER_PASSWORD"
      {{- end }}
      {{- if (include "kafka.saslClientSecretsEnabled" .) }}
      replace_placeholder "controller-client-secret-placeholder" "$KAFKA_CONTROLLER_CLIENT_SECRET"
      {{- end }}
      {{- end }}
      {{- if (include "kafka.client.saslEnabled" .)}}
      read -r -a passwords <<<"$(tr ',;' ' ' <<<"${KAFKA_CLIENT_PASSWORDS:-}")"
      for ((i = 0; i < ${#passwords[@]}; i++)); do
          replace_placeholder "password-placeholder-${i}" "${passwords[i]}"
      done
      {{- end }}
      {{- if .Values.sasl.zookeeper.user }}
      replace_placeholder "zookeeper-password-placeholder" "$KAFKA_ZOOKEEPER_PASSWORD"
      {{- end }}
    }
    {{- end }}
```

kafka.saslEnabled checks only for SASL in listeners but not for zookeeper connection:
https://github.com/bitnami/charts/blob/main/bitnami/kafka/templates/_helpers.tpl#L125-L146

```
{{/*
Return true if SASL connections should be configured
*/}}
{{- define "kafka.saslEnabled" -}}
{{- $res := "" -}}
{{- if (include "kafka.client.saslEnabled" .) -}}
{{- $res = "true" -}}
{{- else -}}
{{- $listeners := list .Values.listeners.interbroker -}}
{{- if and .Values.kraft.enabled -}}
{{- $listeners = append $listeners .Values.listeners.controller -}}
{{- end -}}
{{- range $listener := $listeners -}}
{{- if regexFind "SASL" (upper $listener.protocol) -}}
{{- $res = "true" -}}
{{- end -}}
{{- end -}}
{{- end -}}
{{- if $res -}}
{{- true -}}
{{- end -}}
{{- end -}}
```

**Problem 2**:
In my opinion there is no property `sasl.jaas.config` for brokers in server.properties file:
https://kafka.apache.org/documentation/#brokerconfigs_sasl.jaas.config

>  For brokers, the config must be prefixed with listener prefix and SASL mechanism name in lower-case. For example, listener.name.sasl_ssl.scram-sha-256.sasl.jaas.config=com.example.ScramLoginModule required;

See also:
https://kafka.apache.org/documentation/#security_jaas_broker

So if I include a test kafka_jaas.conf file as workaround with following content sasl connection to zookeeper works as expected:
```
Client {
  org.apache.kafka.common.security.plain.PlainLoginModule required
  username="zookeeper-user"
  password="replaced password";
}
```

In order to create a proper PR I need to discuss the following points:
- Am I right that we need kafka_jaas.conf file instead of sasl.jaas.config for external zookeeper sasl authentication?
- I would say it would be better to use `{{- if .Values.sasl.zookeeper.user }}` as condition for creating zookeeper config and password replacement in a separate code block instead of `{{- if (include "kafka.saslEnabled" .) }}` . Any objections?

### PR DESCRIPTION
**Pull request by ph311o**
### Description of the change

SASL to external Zookeeper was not proper configured.

### Benefits

It is now possible to run Kafka without SASL (e.g. with SSL ACLs) but connect to SASL enabled Zookeeper. The jaas configuration for SASL-Zookeeper was wrong and is fixed.

### Possible drawbacks

n/a

### Applicable issues

- fixes #22729

### Additional information

n/a

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/kafka/templates/scripts-configmap.yaml
Comment (rafariossaa) [inline bitnami/kafka/templates/scripts-configmap.yaml:336]: Hi,
  I think there is a mismatch of the paths here.
  This block generates the file `/config/kafka_jass.conf` but this file no mounted by the pod.
   To mimic what is being done to `/config/server.properties` you would need to copy the file.
   Also, have into account that `KAFKA_CONFIG_FILE` is being overridden in the next line.

Comment (ph311o) [reply bitnami/kafka/templates/scripts-configmap.yaml:336]: No there is no path-mismatch as the initcontainer is mounting kafka-config volume to /config. Later the statefulset is using volume kafka-config to mount server.properties and kafka_jaas.conf files to folder /opt/bitnami/kafka/config/

See https://github.com/bitnami/charts/blob/main/bitnami/kafka/templates/_helpers.tpl#L920
and for example for server.properties mounting:
https://github.com/bitnami/charts/blob/main/bitnami/kafka/templates/broker/statefulset.yaml#L302

The KAFKA_CONFIG_FILE Variable is used in function "replace_placeholder" to find the right file where we have to replace placeholder.

See https://github.com/bitnami/charts/blob/main/bitnami/kafka/templates/scripts-configmap.yaml#L140

Comment (ph311o) [reply bitnami/kafka/templates/scripts-configmap.yaml:336]: Hi,
could you please re-review my PR?
Thank you

Comment (rafariossaa) [reply bitnami/kafka/templates/scripts-configmap.yaml:336]: Sorry for the delay.
Let me take a look

**Review by rafariossaa — DISMISSED** [review]
Thanks for the fix.
LGTM

**Comment by ph311o** [comment]
> Thanks for the fix. LGTM

Thank you. I think you have to re-approve after merging main into branch.

Cheers

**Review by rafariossaa — APPROVED** [review]

**Pull request merged**
